pkg/logentry/stages/metrics.go

```diff
@@ -58,6 +62,24 @@ func validateMetricsConfig(cfg MetricsConfig) error {
 			config.MetricType != MetricTypeHistogram {
 			return errors.Errorf(ErrMetricsStageInvalidType, config.MetricType)
 		}
+
+		// Set the idle duration for metrics
+		if config.IdleDuration != nil {
+			d, err := time.ParseDuration(*config.IdleDuration)
+			if err != nil {
+				return errors.Errorf(ErrInvalidIdleDur, err)
+			}
+			if d < 1*time.Second {
+				return errors.New(ErrSubSecIdleDur)
+			}
+			cp := cfg[name]
+			cp.maxIdleSec = int64(d.Seconds())
+			cfg[name] = cp
+		} else {
+			cp := cfg[name]
+			cp.maxIdleSec = int64(5 * time.Minute.Seconds())
```

Comment:
It would be nice if this default was handle by the config flag like everything else ?

Reply:
there isn't any flags here, nothing like register flags for pipelines.

Rather than creating a separate variable we could just set the duration on the object if it's not provided?

Reply:
ok, now i remember why I did it this way, if you try to use a single object you are forced to essentially parse it into a duration everywhere it gets used.

Creating a non-exposed `maxIdleSec` allows for parsing of the string provided via the config only one time and being able to report any parsing errors in the appropriate place